org.eclipse.transformer/src/main/java/org/eclipse/transformer/action/impl/SignatureRuleImpl.java

```diff
@@ -181,6 +197,37 @@ public BundleData getBundleUpdate(String symbolicName) {
 	public String getDirectString(String initialValue) {
 		return directStrings.get(initialValue);
 	}
+        
+        private final Map<String, Map<String, String>> perClassConstantStrings;
+
+        @Override
+        public String getConstantString(String initialValue, String clazz) {
+            Map<String, String> m = perClassConstantStrings.get(clazz);
+            if (m != null) {
```

Comment:
This might immediately return (with null) if no class specific table ('m') is retrieved.  That would remove a level of indentation from the method, and would de-couple the "no substitution because no table was found" case from the "no substitution because no keys were found in the input string" case.  But, not all folks agree with the style, so I'm OK with this the way it is.

Reply:
I like it better, I made the change you proposed.